GithubJobApp/GithubApiManager.swift

```diff
@@ -7,63 +7,132 @@
+                
+                var jobArray = [Job]()
+                let json: [AnyObject]?
+                do {
+                    json = try NSJSONSerialization.JSONObjectWithData(jobs, options: NSJSONReadingOptions()) as? [AnyObject]
+                    
+                    if let json = json {
+                        print("how many jobs?: \(json.count)")
+                        for job in json {
+                            guard let jobJson = job as? [String: AnyObject] else {
+                                print("error reading job: \(jobArray.count)")
+                                continue
+                            }
+                            
+                            jobArray.append(Job(withDictionary: jobJson))
+                        }
                     }
-                    let companyLogoUrl = jobsJson["company_logo"] as? String
-                    let companyUrl = jobsJson["company_url"] as? String
-                    // Can remove the check for nil because the Job has an optional companyLogoUrl variable
-                    jobArray.append(Job(title: title, company: company, description: description, companyLogoUrl: companyLogoUrl, companyUrl: companyUrl))
                     
-        //                if(companyLogoUrl != nil) {
-        //                    jobArray.append(Job(title: title, company: company, description: description, companyLogoUrl:companyLogoUrl))
-        //                } else {
-        //                    jobArray.append(Job(title: title, company: company, description: description, companyLogoUrl: ""))
-        //                }
-                    print(title + " " + company)
+                    dispatch_async(dispatch_get_main_queue()) {
+                        self.githubAPIDelegate?.jobsRetrieved(jobArray)
+                    }
+                } catch {
+                    print(error)
                 }
             }
-            
-            setDataFunction(jobArray)
+            task.resume()
         }
+    }
+    
+    func generateURL(searchQuery: String?, locationCoordinates: CLLocationCoordinate2D?) -> String {
+        var url = GithubApiManager.baseURL
         
-        task.resume()
+        if searchQuery != nil || locationCoordinates != nil {
+            url.appendContentsOf("?")
         }
-        
+        if let searchQuery = searchQuery {
+            url.appendContentsOf("description=\(searchQuery)")
+            if let locationCoordinates = locationCoordinates {
+                url.appendContentsOf("&lat=\(locationCoordinates.latitude.description)&long=\(locationCoordinates.longitude.description)")
+            }
+        } else {
+            if let locationCoordinates = locationCoordinates {
+                url.appendContentsOf("lat=\(locationCoordinates.latitude.description)&long=\(locationCoordinates.longitude.description)")
+            }
+        }
+        return url
```

Comment:
Foundation has classes available for building URL strings.  They take care of escaping special characters and keep track of all the ?s and &s. 

```swift
let urlComps = NSURLComponents(string: "https://api.github.com/baz/")
var qItems = [URLQueryItem]()
qItems.append(URLQueryItem(name: "foo", value: "baz"))
qItems.append(URLQueryItem(name: "lunch", value: "fish and chips"))
urlComps?.queryItems = qItems

let finalUrl = urlComps?.url
```

NSURLCompoents is also good for parsing a given URL, if you need to go the other way.